### ISSUE TITLE
Filetree read does not have the gateway objects

### ISSUE THREAD

**Issue by bogdanmuresan** [redhat-cop/aap_configuration_extended#44]
## Summary
Filetree read does not have the gateway objects

## Issue Type

- Bug Report

**Comment by bogdanmuresan** [comment]
Is this really not planned, or closed by accident?

**Comment by ivarmu** [comment]
Closed for being stale I think... I'm working on it.

**Comment by bogdanmuresan** [comment]
Hello @silvinux and @ivarmu  - this issue was created for filetree_read - the objects were only added to filetree_create.

**Comment by ivarmu** [comment]
Yeah, but `filetree_read` is also able to read the gateway and EDA objects

**Comment by bogdanmuresan** [comment]
I don;t see any code in there that would do that.. 
https://github.com/redhat-cop/aap_configuration_extended/tree/devel/roles/filetree_read

Or am I missing something?

**Comment by ivarmu** [comment]
> I don;t see any code in there that would do that.. https://github.com/redhat-cop/aap_configuration_extended/tree/devel/roles/filetree_read
> 
> Or am I missing something?

`filetree_read` is only reading all the variables defined in the input files, recursively. The main difference from loading these variables from `group_vars` and using `filetree_read` is that the latter allows to have more than one file defining the same variable, so `filetree_read` will collapse all these repeated variables into one final variable containing all the contents that are spread through the different input files. In the case of the Gateway and EDA objects, `filetree_create` is not creating more than one file per each object, so there's (still) no need to explicitly collapse any repeated input variable, and that's why there are no explicit `tasks/*` files to do so (by now).

Hope this clarifies your doubts.

**Comment by bogdanmuresan** [comment]
Hi @ivarmu , and thank you for your answer.  We kind of wished we had our config as code file structure consistent (under "config/org/env") without any other changes. Not a big deal, I guess we can use "ansible.builtin.include_vars" in the playbook to include these files. Thank you again!

**Comment by lennysh** [comment]
Is this now solved with #179 ?
